lib/cocoapods-downloader/git.rb

```diff
@@ -22,6 +22,23 @@ def checkout_options
         end
       end
 
+      def self.preprocess_options(options)
+        return options unless options[:branch]
+
+        command = ['ls-remote',
+                   options[:git],
+                   options[:branch]]
+        output = Executable.execute_command('git', command)
```

Comment:
ah silly me. we'll figure out how to make that work :) In the meantime, you can do something hacky that'll let you start writing tests / playing around with it if you want